### DISCUSSION TITLE
How to write only the EEPROM on a attiny202

### DISCUSSION THREAD

**Discussion by NimbleMotorsports** [SpenceKonde/megaTinyCore#1098] [Library discussion]
It does not look there is an option in the Arduino IDE.
It seems I need to use a python script of some kind, but I have not found the answer via google.  Any help appreciated.   I was editing my code for each device to change its ID, but that is just dumb.  However, so far I could have done that 4 times over for all the time I've spent trying to find out to write it into the eeprom..

**Comment by mechatroniks-git** [comment]
You can use `pymcuprog` on cli to read & write. Example

`
pymcuprog -U -t uart -u COM8 -d attiny1614 write -m user_row -l ...`

Read contents

`pymcuprog -t uart -u COM8 -d attiny1614 read -m eeprom`

Erase
`pymcuprog -t uart -u COM8 -d attiny1614 erase -m user_row`

**Comment by NimbleMotorsports** [comment]
Just what I needed to know, surprising how hard it was to find this info.
   
  
   I ran into another problem...that you must use -l to specify a byte value to write as the last option,
   
  
   otherwise anything after -l is parsed as values to write and not commands or switches to the command.
   
  
   Hope that helps someone in the future.  So this fails, pymcuprog -o 0 -b 1 -t 0x22 -d attiny202 -m eeprom write
   
  
   it must be pymcuprog write -o 0 -b 1 -d attiny202 -m eeprom -t 0x22
   
  
    
   
  
   On 05/09/2024 6:17 AM PDT Mechatroniks ***@***.***> wrote:
   
   
   
     
    
   
     
    
     
   You can use 'pymcuprog' on cli to read & write. Example 
   'pymcuprog -U -t uart -u COM8 -d attiny1614 write -m user_row -l ...'

**Comment by NimbleMotorsports** [comment]
How difficult would it be to add something to the arduino ide to do this?
   
  
   The IDE has an extensions option in the toolbar.  If I can help contribute to add that..
   
  
   I had to install python and install pymcuprog (windows 10) but we know it already was there in the megaTinyCore,
   
  
   which I tried to find and was not obvious.   If I knew how to invoke it from there I could add the extension.
   
   
   
    On 05/09/2024 6:17 AM PDT Mechatroniks ***@***.***> wrote:
    
   
     
    
   
     
    
     
   You can use 'pymcuprog' on cli to read & write. Example 
   'pymcuprog -U -t uart -u COM8 -d attiny1614 write -m user_row -l ...'

**Reply by mechatroniks-git** [reply]
Not sure about how to do something like that in Arduino, but Platform IO allows you to make a pre downloaded and/or post upload call that easily could do this.
